### PR TITLE
CCivilianPed done & restore peds running to phone

### PR DESCRIPTION
**Pull request by erorcun**

### CODE REVIEW

File: src/peds/CivilianPed.cpp [hunk 1]
```diff
@@ -2,25 +2,420 @@
...
+	for (int i = 0; i < ARRAY_SIZE(m_nearPeds); i++) {
 		m_nearPeds[i] = nil;
 	}
 }
 
+void
+CCivilianPed::CivilianAI(void)
+{
+#ifdef TOGGLEABLE_BETA_FEATURES
+	if (bRunningToPhone && m_nPedState != PED_SEEK_POS && m_nPedState != PED_FALL && m_nPedState != PED_GETUP &&
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: `!a&&!b` -> `a||b`

File: src/peds/CivilianPed.cpp [hunk 2]
```diff
@@ -2,25 +2,420 @@
...
+
+		} else if (closestThreatFlag == PED_FLAG_GUN) {
+			SetLookFlag(m_threatEntity, true);
+			SetLookTimer(500);
+		}
+		return;
+	}
+	uint32 closestThreatFlag = ScanForThreats();
+	if (closestThreatFlag == PED_FLAG_GUN) {
+		if (!m_threatEntity || !m_threatEntity->IsPed())
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: also?

File: src/peds/CivilianPed.cpp [hunk 3]
```diff
@@ -2,25 +2,420 @@
...
+						SetFlee(m_threatEntity, 10000);
+						bUsePedNodeSeek = true;
+						m_pNextPathNode = nil;
+					}
+				}
+			} else if (m_objective != OBJECTIVE_NONE || GetWeapon()->IsTypeMelee()) {
+				SetFlee(m_threatEntity, 5000);
+				bUsePedNodeSeek = true;
+				m_pNextPathNode = nil;
+				if (threatDist >= 400.0f) {
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: maybe better `threatDist >= 400.0f` and flip code?

File: src/peds/CivilianPed.cpp [hunk 4]
```diff
@@ -2,25 +2,420 @@
...
+
+		} else {
+#ifdef TOGGLEABLE_BETA_FEATURES
+			CEntity* killerOfDeadPed = ((CPed*)m_pEventEntity)->m_threatEntity;
+			eCrimeType crime = (((CPed*)m_pEventEntity)->m_ped_flagI40 ?
+				(((CPed*)m_pEventEntity)->m_nPedType == PEDTYPE_COP ? CRIME_RUNOVER_COP : CRIME_RUNOVER_PED) :
+				(((CPed*)m_pEventEntity)->m_nPedType == PEDTYPE_COP ? CRIME_SHOOT_COP : CRIME_SHOOT_PED));
+			bool eligibleToReport = bMakePedsRunToPhonesToReportCrimes && killerOfDeadPed && killerOfDeadPed->IsPed() && ((CPed*)killerOfDeadPed)->IsPlayer() &&
+				m_pedStats->m_fear <= m_pedStats->m_lawfulness;
+			if(!eligibleToReport || !RunToReportCrime(crime))
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: also flip?

File: src/peds/CivilianPed.cpp [hunk 5]
```diff
@@ -2,25 +2,420 @@
...
+						RestorePreviousState();
+						m_phoneId = -1;
+#ifdef FIX_BUGS
+						bRunningToPhone = false;
+#endif
+					} else {
+						gPhoneInfo.m_aPhones[m_phoneId].m_nState = PHONE_STATE_REPORTING_CRIME;
+						m_nPedState = PED_FACE_PHONE;
+					}
+				} else if (m_objective != OBJECTIVE_KILL_CHAR_ANY_MEANS && m_objective != OBJECTIVE_KILL_CHAR_ON_FOOT) {
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: also

File: src/peds/CivilianPed.cpp [hunk 6]
```diff
@@ -2,25 +2,420 @@
...
+				SetFlee(m_pEventEntity, 5000);
+				bUsePedNodeSeek = true;
+				m_pNextPathNode = nil;
+				SetMoveState(PEDMOVE_RUN);
+			}
+		}
+	} else if (closestThreatFlag == PED_FLAG_EXPLOSION) {
+		CVector2D eventDistVec = m_eventOrThreat - GetPosition();
+		float eventDist = eventDistVec.MagnitudeSqr();
+		if (eventDist >= 400.0f) {
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: also flip?

File: src/peds/Ped.cpp [hunk 7]
```diff
@@ -3168,6 +3173,15 @@ CPed::CheckAroundForPossibleCollisions(void)
 bool
 CPed::MakePhonecall(void)
 {
+#ifdef TOGGLEABLE_BETA_FEATURES
+	if (bMakePedsRunToPhonesToReportCrimes)
+		if (!IsPlayer() && CTimer::GetTimeInMilliseconds() > m_phoneTalkTimer - 7000 && bRunningToPhone) {
+
+			FindPlayerPed()->m_pWanted->RegisterCrime_Immediately(m_crimeToReportOnPhone, GetPosition(),
+				(m_crimeToReportOnPhone == CRIME_POSSESSION_GUN ? (int)m_threatEntity : (int)((CPed*)m_pEventEntity)->m_threatEntity), false);
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: i'd extract condition and add extra variable for entity.

File: src/peds/CivilianPed.cpp [hunk 8]
```diff
@@ -2,25 +2,420 @@
...
+	} else if (closestThreatFlag == PED_FLAG_DEADPEDS) {
+		float eventDist = (m_pEventEntity->GetPosition() - GetPosition()).MagnitudeSqr2D();
+		if (IsGangMember() && m_nPedType == ((CPed*)m_pEventEntity)->m_nPedType) {
+			if (eventDist < 25.0f) {
+				SetFlee(m_pEventEntity, 2000);
+				bUsePedNodeSeek = true;
+				m_pNextPathNode = nil;
+				SetMoveState(PEDMOVE_RUN);
+			}
+		} else if (IsGangMember() || eventDist > 25.0f) {
```
Comment (ShFil119) [inline src/peds/CivilianPed.cpp]: can you recheck this float comparison, looks a bit suspicious.

File: src/peds/CivilianPed.cpp [hunk 9]
```diff
@@ -2,25 +2,420 @@
...
+			if (CTimer::GetTimeInMilliseconds() > CWorld::Players[CWorld::PlayerInFocus].m_nNextSexFrequencyUpdateTime) {
+				if (m_nPedState == PED_DRIVING
+					&& m_pMyVehicle->pDriver && m_pMyVehicle->pDriver->IsPlayer() && m_pMyVehicle->pDriver->m_nPedState == PED_DRIVING) {
+					CColPoint foundCol;
+					CEntity* foundEnt;
+
+					CWorld::ProcessVerticalLine(m_pMyVehicle->GetPosition(), -100.0f,
+						foundCol, foundEnt, true, false, false, false, false, false, nil);
+
+					if (m_pMyVehicle->m_vecMoveSpeed.MagnitudeSqr() >= 0.0001f
```
Comment (Nick007J) [inline src/peds/CivilianPed.cpp]: SQR(0.01f)

File: src/peds/Ped.cpp [hunk 10]
```diff
@@ -12474,7 +12524,7 @@ CPed::ProcessObjective(void)
 								m_pMyVehicle->m_status = STATUS_PHYSICS;
 								m_pMyVehicle->AutoPilot.m_nPrevRouteNode = 0;
 								if (m_nPedType == PEDTYPE_COP) {
-									m_pMyVehicle->AutoPilot.m_nCruiseSpeed = (CWorld::Players[CWorld::PlayerInFocus].m_pPed->m_pWanted->m_nWantedLevel * 0.1f + 0.6f) * (60.0f * m_pMyVehicle->pHandling->Transmission.fUnkMaxVelocity);
+									m_pMyVehicle->AutoPilot.m_nCruiseSpeed = (FindPlayerPed()->m_pWanted->m_nWantedLevel * 0.1f + 0.6f) * (60.0f * m_pMyVehicle->pHandling->Transmission.fUnkMaxVelocity);
```
Comment (Nick007J) [inline src/peds/Ped.cpp]: 60.0f is GAME_SPEED_TO_CARAI_SPEED (from CarCtrl.h)

File: src/peds/CivilianPed.cpp (hunk 1 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: I think there is no problem in here, aap only against the ones with 'else' becuase he thinks they're originally written with a || b and we should flip the if/else.

File: src/peds/CivilianPed.cpp (hunk 2 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: aap probably more ok with that since it's an early return.

File: src/peds/CivilianPed.cpp (hunk 4 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: again there is no else and this isn't game's code afterall, so there is no coding style we should guess how it was written

File: src/peds/CivilianPed.cpp (hunk 5 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: again, I don't think there is a problem here :/ but below here there was that
```} else if (m_objective != OBJECTIVE_GOTO_AREA_ON_FOOT && m_objective != OBJECTIVE_RUN_TO_AREA) {
} else {
...
}```

and I'll flip it in next update

File: src/peds/CivilianPed.cpp (hunk 2 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: I think same as above

File: src/peds/CivilianPed.cpp (hunk 3 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: ok

File: src/peds/CivilianPed.cpp (hunk 6 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: ok

File: src/peds/Ped.cpp (hunk 7 above)
Comment (erorcun) [reply src/peds/Ped.cpp]: I don't wanna do that unless it's strictly forced by some code style we follow or by someone :/

File: src/peds/CivilianPed.cpp (hunk 8 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: it's correct, but probably I will refactor it to make it look better

File: src/peds/Ped.cpp (hunk 10 above)
Comment (erorcun) [reply src/peds/Ped.cpp]: thanks, done

File: src/peds/CivilianPed.cpp (hunk 9 above)
Comment (erorcun) [reply src/peds/CivilianPed.cpp]: ok

**Pull request merged**